src/openrct2/actions/PeepPickupAction.cpp

```diff
@@ -23,6 +23,15 @@ PeepPickupAction::PeepPickupAction(PeepPickupType type, EntityId spriteId, const
 {
 }
 
+
+void PeepPickupAction::AcceptParameters(GameActionParameterVisitor& visitor)
+{
+    visitor.Visit("type", _type);
+    visitor.Visit("entityId", _spriteId);
+    visitor.Visit(_loc);
+    visitor.Visit("playerId", _owner);
+}
```

Comment:
Have you tried to use this action I remember that the way it works is a bit finickity I am unsure if there is enough information exposed to plugins to use it properly from plugins.

Reply:
Yes, I tested every action (although ofc not every parameter combination).
It is not nice to use, but it works. When picking up, I could not manage to make the UI show it. I.e. let the plugin pick up and the player drop it. But otherwise it's fine. Although it is annoying that you need to provide the original `x` value when you cancel.

Reply:
okay then.